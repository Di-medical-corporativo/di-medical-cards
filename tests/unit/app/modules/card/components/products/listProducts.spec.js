import { shallowMount } from '@vue/test-utils'
import ListProduct from '../../../../../../../src/app/modules/card/components/products/ListProducts.vue'

jest.mock('vue-router', () => ({
  useRouter: jest.fn(() => ({
    push: jest.fn()
  })),
  useRoute: jest.fn(() => ({
    params: {
      sucursal: 'sur'
    }
  }))
}))

const mockStore = {
  dispatch: jest.fn().mockReturnValue(null),
  commit: jest.fn(),
  getters: ['']
}

jest.mock('vuex', () => ({
  useStore: () => mockStore
}))

describe('List product component', () => {
  beforeEach(() => jest.clearAllMocks())

  test('should match snapshot', () => {
    const wrapper = shallowMount(ListProduct)

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should dispatch action when component is created', () => {
    shallowMount(ListProduct)
    expect(mockStore.dispatch).toHaveBeenCalledTimes(1)
  })
})
