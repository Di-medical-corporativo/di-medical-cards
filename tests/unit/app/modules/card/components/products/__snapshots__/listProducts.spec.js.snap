// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List product component should match snapshot 1`] = `<div>cargando...</div>`;
